### ISSUE TITLE
Feature: Split markdown shortcuts, import and export

### ISSUE THREAD

**Issue by fantactuka** [facebook/lexical#4550]
Current `@lexical/markdown` package supports shortcuts (auto-replace while typing), import and export relying on the same transformers configuration. It also uses RegExp parsing that does not have any external dependencies, which is great for bundle size. On the other hand it has few disadvantages:
- If app only uses shortcuts (no import/export), transformers would still require all import/export related properties
- Transformers API became quite bloated with more expansion requests (#4500)
- RegExp matching is nowhere near to be as good as proper tokenization (#2632, #2214)

Related issues/PRs: #4449, #4500, #4339, #4461, and [more](https://github.com/facebook/lexical/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Amarkdown)

I think we can split these into separate features (or even packages), where shortcuts will keep relying on lightweight RegExp matching and simplified transformers configuration; import/export will use 3rd party dependency for proper tokenization (these features are less common and also can be lazy-loaded for certain scenarios, keeping initial bundle small)

// cc @themagickoala @xinyuan0801 @hanford would appreciate if you guys have any suggestions/ideas (e.g., regarding 3rd party parser) given you've had PPs for Lexical MD

**Comment by themagickoala** [comment]
@fantactuka thanks for thinking of me! https://github.com/themagickoala/lexical-remark is using the remark ecosystem for MD parsing, and it's working a treat. I was able to add video support in a day, and just released collapsible container support in 2 days. These were things that the playground wasn't supporting for import/export, so it was nice to be able to get done!

I can understand the approach you've taken with RegExp, not wanting to add dependencies to the project. Once you start going down that 3rd party route, it definitely bloats things a little. If you do end up being keen on remark, I'm happy to consult as I've become very familiar with it over the past few weeks 😆.

Supporting a separate shortcuts system would be a great idea. I've not started looking at shortcuts in lexical-remark, so I can't give much advice there, but I think RegExp is probably sufficient for matching while typing!

**Comment by vadimkantorov** [comment]
Yeah, plugging a reliable markdown importer/exporter via a widely-used package (like remark?) should be better... Currently many small issues exist with Lexical's support for markdown like https://github.com/facebook/lexical/issues/7234 - likely stemming from the current regexp-based approach and focus on simplicity...

**Comment by vadimkantorov** [comment]
Also, if remark can export to a more flexible variant of markdown with some HTML sprinkled, it should enable support of more rich-text like colored text, detail-summary etc 

This is important usecase for WYSIWYG rich-text editing for CMS: mainly use markdown when we can, and fallback to HTML still achieving export (or import from such extended Markdown)

E.g. quite some HTML is supported in Markdown rendering on GitHub

**Comment by etrepum** [comment]
`@lexical/mdast` implements shortcuts, import and export with the same plumbing that remark uses - there's a bundle size cost of about 30kb gzipped but it's much more correct #8794
